In [43]:
import numpy as np
import pandas as pd

In [44]:
dataframe = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### ACTIVIDAD 1:

#### ¿Quién produjo originalmente los datos?<br>
World Population Prospects
#### ¿Qué organización los procesa para su utilización en Our World in Data?<br>
Our World in Data
#### ¿Qué representa la variable que analizarás?<br>
el crecimiento anual de la poblacion en los paises


In [45]:
dataframe.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [46]:
dataframe.shape

(38400, 5)

In [47]:
dataframe.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [48]:
dataframe.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [49]:
dataframe.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [50]:
dataframe.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [51]:
dataframe["Entity"].nunique() 

256

### ACTIVIDAD 2:
#### ¿Cuántas filas y columnas tiene el dataset?
38400 filas y 5 columnas

#### ¿Qué tipo de datos contiene la variable Year?
int64

#### ¿Existen valores faltantes?
si, 1200 de code, 19712 de Annual population change y 18688 en Annual popolation change (projected)

#### ¿Cuántas entidades diferentes contiene el dataset?
256

In [52]:
df_2019 = dataframe[dataframe['Year'] == 2019]
poblacion_2019 = df_2019['Annual change in population']

entidad_min = df_2019.loc[poblacion_2019.idxmin()]["Entity"]
entidad_max = df_2019.loc[poblacion_2019.idxmax()]["Entity"]

print("media =", poblacion_2019.mean())
print("mediana =", poblacion_2019.median())
print("varianza =", poblacion_2019.var())
print("desviacion_std =", poblacion_2019.std())
print("minimo =", poblacion_2019.min(), (entidad_min))
print("maximo =", poblacion_2019.max(), (entidad_max))

media = 2261236.08984375
mediana = 43636.5
varianza = 99503089007422.66
desviacion_std = 9975123.508379364
minimo = -869495.0 Venezuela
maximo = 81390920.0 World


### ACTIVIDAD 3:
#### ¿Cuál es el valor de la media?
Es de 2261236.08984375
#### ¿Cuál es el valor de la mediana?
Es de 43636.5
#### ¿Cuál es el valor de la desviación estándar?
Es de 9975123.508379364
#### ¿Qué entidad presenta el valor mínimo y cuál el máximo?
El valor minimo diria yo que es las personas que mas murieron que en este caso fue en venezuela y supongo que el maximo es las personas que nacieron en ese año

In [53]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)


Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


## Inicio de practica 2


In [54]:
dataframe.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN


In [55]:
#Series df_year =dataframe["Year"]

df_year =dataframe[["Year", "Entity"]]
df_year.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [56]:
df_extraccion_columnas =dataframe[["Year", "Entity"]]
df_extraccion_columnas.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [57]:
df_extraccion_filas =dataframe[dataframe["Year"] == 2010]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
59,Afghanistan,AFG,2010,817986.0,NaN
209,Africa (UN),UN_AFR,2010,27438963.0,NaN
359,Albania,ALB,2010,-32756.0,NaN
509,Algeria,DZA,2010,697795.0,NaN
659,American Samoa,ASM,2010,-296.0,NaN


In [61]:
df_2020_mex = dataframe[(dataframe["Entity"] == "Mexico") & ( dataframe["Year"].between(2003,2020))] [["Entity","Year","Annual change in population"]]
df_2020_mex.head() 


,Entity,Year,Annual change in population
22252,Mexico,2003,1429890.0
22253,Mexico,2004,1415614.0
22254,Mexico,2005,1417373.0
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0


In [63]:
df_2020_mex.mean(numeric_only=True)

Year                           2.011500e+03
Annual change in population    1.402801e+06
dtype: float64